src/cavendish_particle_tracks/_main_widget.py

```diff
@@ -789,6 +789,13 @@ def _on_click_save(self) -> None:
             self.msg.show()
             return
 
+        with open(file_name, "w", encoding="UTF8", newline="") as f:
+            # write the header
+            f.write(",".join(self.data[0].vars_to_save()) + "\n")
+
+            # write the data
+            f.writelines([particle.to_csv() for particle in self.data])
```

Comment:
Why is this repeated? This code seems to duplicate the code on line 774ff. ☝️

Reply:
Oops, too sleepy yesterday...